samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferSampleTest.java

```diff
@@ -0,0 +1,241 @@
+    static final String TRANSFER_FILE_PATH = "samples/04.0-file-transfer/filetransfer.json";
+    // Reuse an already existing file for the test. Could be set to any other existing file in the repository.
+    static final String SAMPLE_ASSET_FILE_PATH = TRANSFER_FILE_PATH;
+    static final String DESTINATION_FILE_PATH = "samples/04.0-file-transfer/consumer/requested.test.txt";
+    static final Duration TIMEOUT = Duration.ofSeconds(15);
+    static final Duration POLL_INTERVAL = Duration.ofMillis(1000);
+    static final String API_KEY_HEADER_KEY = "X-Api-Key";
+    static final String API_KEY_HEADER_VALUE = "password";
+    @RegisterExtension
+    static EdcRuntimeExtension provider = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:provider",
+            "provider",
+            Map.of(
+                    // Override 'edc.samples.04.asset.path' implicitly set via property 'edc.fs.config'.
+                    "edc.samples.04.asset.path", getFileFromRelativePath(SAMPLE_ASSET_FILE_PATH).getAbsolutePath(),
+                    "edc.fs.config", getFileFromRelativePath(PROVIDER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath()
+            )
+    );
+    @RegisterExtension
+    static EdcRuntimeExtension consumer = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:consumer",
+            "consumer",
+            Map.of(
+                    "edc.fs.config", getFileFromRelativePath(CONSUMER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath()
+            )
+    );
+    final File destinationFile = getFileFromRelativePath(FileTransferSampleTest.DESTINATION_FILE_PATH);
+    String contractNegotiationId;
+    String contractAgreementId;
+
+    /**
+     * Resolves a {@link File} instance from a relative path.
+     */
+    @NotNull
+    static File getFileFromRelativePath(String relativePath) {
+        return new File(TestUtils.findBuildRoot(), relativePath);
+    }
+
+    /**
+     * Reads a properties file and returns a {@link Properties} instance for the given properties file.
+     * @param fileName Path to a properties file.
+     * @return A {@link Properties} instance for the given properties file.
+     * @throws IOException Thrown if error while accessing the specified properties file.
+     */
+     static Properties readPropertiesFile(String fileName) throws IOException {
+        try (var propertiesFileInputStream = new FileInputStream(fileName)) {
+            Properties prop = new Properties();
+            prop.load(propertiesFileInputStream);
+            return prop;
+        }
+    }
+
+    /**
+     * Run all sample steps in one single test.
+     * Note: Sample steps cannot be separated into single tests because {@link EdcRuntimeExtension}
+     * runs before each single test.
+     */
+    @Test
+    void runSampleSteps() throws IOException {
+        assertTestPrerequisites();
+
+        assertInitiateContractNegotiation();
```

Comment:
why are these methods called `assert` since they do more than asserting? I would remove that prefix